bundles/mapping/shadow-plugin-3d/view/ShadowTool/ShadowTool.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Mutator } from 'oskari-ui/util';
+import { Background, StyledIcon, Row, Col, StyledInput, StyledButton } from './ShadowToolStyled';
+
+export const ShadowTool = ({ mutator, date, time }) => {
+    const [timeValue, setTime] = React.useState(time);
+    const [dateValue, setDate] = React.useState(date);
+    const setCurrentTime = () => {
+        const d = new Date();
+        const curTime = `${d.getHours()}:${d.getMinutes()}`;
+        const curDate = `${d.getDate()}/${d.getMonth() + 1}`;
+        setTime(curTime);
+        setDate(curDate);
+        mutator.setCurrentTime(curDate, curTime);
+    };
+
+    const validateTime = (target) => {
+        const regex = /^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$/;
+        return regex.test(target);
+    };
+
+    const validateDate = (target) => {
+        const matches = /^(0[1-9]|1[0-9]|2[0-9]|3[0-1]|[1-9])[/](0[1-9]|1[0-2]|[1-9])$/.exec(target);
+        if (matches === null) {
+            return false;
+        }
+        const d = parseInt(matches[1]);
+        const m = matches[2] - 1;
+        const dateObject = new Date(2019, m, d);
+        return dateObject.getDate() === d && dateObject.getMonth() === m;
+    };
```

Comment:
Add a comment that these are duplicated in the SetTimeRequestHandler. Perfect option would be to have some util.js on 3d-mapmodule that could be imported and used both here and in the requesthandler but the comment is enough for now.

Reply:
There will be a follow up PR that has changes to this so merging for now.